nri-http/src/Http.hs

```diff
@@ -212,36 +220,102 @@ _request doAnythingHandler manager settings = do
+            Internal.ExpectBytesResponse mkResult -> mkResult (Internal.GoodStatus_ (mkMetadata okResponse) (Data.ByteString.Lazy.toStrict bytes))
+    Left exception ->
+      case expect of
+        Internal.ExpectTextResponse mkResult ->
+          mkResult
+            <| case exception of
+              HTTP.HttpExceptionRequest _ content ->
+                case content of
+                  HTTP.StatusCodeException res bytes ->
+                    let bodyAsText = Data.Text.Encoding.decodeUtf8 bytes
+                     in Internal.BadStatus_ (mkMetadata res) bodyAsText
+                  HTTP.ResponseTimeout ->
+                    Internal.Timeout_
+                  HTTP.ConnectionTimeout ->
+                    Internal.NetworkError_ "ConnectionTimeout"
+                  HTTP.ConnectionFailure err ->
+                    Internal.NetworkError_ (Text.fromList (Exception.displayException err))
+                  err ->
+                    Internal.NetworkError_ (Text.fromList (show err))
+              HTTP.InvalidUrlException _ message ->
+                Internal.BadUrl_ (Text.fromList message)
+        Internal.ExpectBytesResponse mkResult ->
+          mkResult
+            <| case exception of
+              HTTP.HttpExceptionRequest _ content ->
+                case content of
+                  HTTP.StatusCodeException res bytes ->
+                    Internal.BadStatus_ (mkMetadata res) bytes
+                  HTTP.ResponseTimeout ->
+                    Internal.Timeout_
+                  HTTP.ConnectionTimeout ->
+                    Internal.NetworkError_ "ConnectionTimeout"
+                  HTTP.ConnectionFailure err ->
+                    Internal.NetworkError_ (Text.fromList (Exception.displayException err))
+                  err ->
+                    Internal.NetworkError_ (Text.fromList (show err))
+              HTTP.InvalidUrlException _ message ->
+                Internal.BadUrl_ (Text.fromList message)
+        _ ->
+          case exception of
+            HTTP.HttpExceptionRequest _ content ->
+              case content of
+                HTTP.StatusCodeException res _ ->
+                  let statusCode = fromIntegral << Status.statusCode << HTTP.responseStatus
+                   in Err (Internal.BadStatus (statusCode res))
+                HTTP.ResponseTimeout ->
+                  Err Internal.Timeout
+                HTTP.ConnectionTimeout ->
+                  Err (Internal.NetworkError "ConnectionTimeout")
+                HTTP.ConnectionFailure err ->
+                  Err (Internal.NetworkError (Text.fromList (Exception.displayException err)))
+                err ->
+                  Err (Internal.NetworkError (Text.fromList (show err)))
+            HTTP.InvalidUrlException _ message ->
+              Err (Internal.BadUrl (Text.fromList message))
+
+mkMetadata :: HTTP.Response a -> Internal.Metadata
+mkMetadata response =
```

Comment:
🎨 NamedFieldPuns is enabled in package.yaml. Why not pattern match for `reponseStatus` and `responseHeaders` here instead of `let`?

Reply:
nevermind: `Response` is opaque.